src/makemon.c

```diff
@@ -354,6 +354,11 @@ m_initweap(register struct monst *mtmp)
             while (!rn2(3))
                 (void) mongets(mtmp, APPLE + rn2(CARROT - APPLE));
         } else if (is_dwarf(ptr)) {
+            if (!rn2((In_mines(&u.uz)) ? 40 : 8)) {
+                otmp = mongets(mtmp, POT_BOOZE);
+                otmp->quan = rnd(3);
+                otmp->owt = weight(otmp);
```

Comment:
i included this "otmp->owt" line simply because other makemon entries that add multiple items tend to also have this line.  i assume it updates the weight of a stack of objects?  i notice it isn't included when mongets flint.

is there a thesaurus for all these hyper-abbreviated terms like otmp mtmp owt etc?

Reply:
Not all, but a lot of fields in structs like `mon` and `obj` have comments explaining what they represent in the header files where the structs are defined (e.g. include/mon.h). There are some explanations [in here](https://nethackwiki.com/wiki/Beginner%27s_guide_to_NetHack_sources) as well, though quite a few parts of that textfile are outdated by now. `owt` doesn't have a comment but you're right that it holds the object's (or stack's) weight.

It's probably a bug that `owt` is not set for the case where hobbits are generated with rocks/flint. I just tested in wizmode and a hobbit was generated with a stack of 7 flint stones that weighed only 2 au.